windows/ReactTestApp/MainPage.cpp

```diff
@@ -6,27 +6,85 @@
 #include "MainPage.g.cpp"
 #include "Manifest.h"
 
-using winrt::Windows::Foundation::Collections::IVector;
+using namespace winrt::Microsoft::ReactNative;
+using namespace winrt::Windows::UI::Xaml::Controls;
 
 namespace winrt::ReactTestApp::implementation
 {
     MainPage::MainPage()
     {
         InitializeComponent();
+        SetComponents();
+        InitReact();
     }
 
-    IVector<ReactTestApp::ComponentViewModel> MainPage::Components()
+    void MainPage::SetComponents()
     {
-        IVector<ReactTestApp::ComponentViewModel> components =
-            winrt::single_threaded_observable_vector<ReactTestApp::ComponentViewModel>();
-
+        auto menuItems = MenuFlyout().Items();
         for (auto &&c : ::ReactTestApp::GetManifest().components) {
-            hstring componentName = to_hstring(c.displayName.value_or(c.appKey));
+            hstring componentDisplayName = to_hstring(c.displayName.value_or(c.appKey));
+            hstring componentName = to_hstring(c.appKey);
             ReactTestApp::ComponentViewModel newComponent =
-                winrt::make<ComponentViewModel>(componentName);
-            components.Append(newComponent);
+                winrt::make<ComponentViewModel>(componentName, componentDisplayName);
+            m_components.push_back(newComponent);
+
+            MenuFlyoutItem newMenuItem = MenuFlyoutItem();
```

Comment:
Having `= MenuFlyoutItem()` here is redundant. The constructor is always called with just the instantiation.
```suggestion
            MenuFlyoutItem newMenuItem;
```